pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -499,17 +499,34 @@ func (d *depsCollector) fetchDependencies(ctx context.Context, purl string, docC
 	component.DepPackages = append(component.DepPackages, dependencyNodes[1:]...)
 
 	for _, edge := range deps.Edges {
-		isDep := &model.IsDependencyInputSpec{
-			VersionRange:   edge.Requirement,
-			DependencyType: model.DependencyTypeDirect,
-			Justification:  "dependency data collected via deps.dev",
-		}
 		foundDepPackage := &IsDepPackage{
 			CurrentPackageInput: dependencyNodes[edge.FromNode].CurrentPackage,
 			DepPackageInput:     dependencyNodes[edge.ToNode].CurrentPackage,
-			IsDependency:        isDep,
+			IsDependency: &model.IsDependencyInputSpec{
+				VersionRange: edge.Requirement,
+				// direct dependency from to the to node. See issue: https://github.com/google/deps.dev/issues/12#issuecomment-1517103380
+				// for more details and example
+				DependencyType: model.DependencyTypeDirect,
+				Justification:  "dependency data collected via deps.dev",
+			},
 		}
 		component.IsDepPackages = append(component.IsDepPackages, foundDepPackage)
+
+		// if its not from the root node, it is an indirect dependency. See issue: https://github.com/google/deps.dev/issues/12#issuecomment-1517103380
```

Comment:
Actually I am not sure if this adds value. What do the others think? As based on the graph we can determine that it's an indirect or direct dependency from the root node.